packages/data-transport-layer/src/services/l2-ingestion/service.ts

```diff
@@ -109,6 +126,8 @@ export class L2IngestionService extends BaseService<L2IngestionServiceOptions> {
 
         await this.state.db.setHighestSyncedUnconfirmedBlock(targetL2Block)
 
+        this.l2IngestionMetrics.highestSyncedL2Block.set(targetL2Block)
```

Comment:
Are there problems here if `this.metrics` is `false`? Will it result in a `cannot call set on undefined` error?

Reply:
^ohh this is a good catch, i think it probably would

Reply:
Changed it to make `metrics` mandatory